### PR TITLE
lib/db: Catch unignored/conflicting files as needed (fixes #5053)

### LINKED ISSUE #5053: Unignored file not considered needed
@AudriusButkevicius @calmh This is yet another issue introduced in https://github.com/syncthing/syncthing/pull/4958 and thus 0.14.49 material - I am really sorry about that disaster and slow piece-by-piece resolution. It also needs another db transition while there is already one on master, that's not in release - I will file separate PRs for master and release branches to address that. Regarding release schedules: May it be reasonable to delay 0.14.50, but already release 0.14.51-rc.1 as normal?

Originally came up in https://forum.syncthing.net/t/did-something-recently-changed-in-ignores-handling/11714/35

Steps to reproduce:

1. Share a folder between devices A and B.
2. Modify file F on A, wait for sync.
3. Modify file F on B, wait for sync.
4. Ignore F on A.
5. Edit F on A and B.
6. Unignore F on A.

Expected: Create and sync conflicts.

Actual: Nothing is synced and no conflicts are created.

### PR DESCRIPTION
**Pull request by imsodin**
### Purpose

Fixes #5053 by doing the check, if the globally updated file needs to be added/removed from the need bucket, before updating the meta counter. Thus it happens on every update, not only if the first item in the global list is affected by the update. An update can be relevant to what's locally needed without happening at the front of the global list, if it involves conflicting files.

This also requires a recalculation of the need bucket. As the local flags change was introduced in dbVersion 5, but is not yet introduced in the release branch, dbVersion 5 needs to be skipped. The local flags change now happens to 7 (for master), while the need bucket reset happens to 6 (for release). Thus even if someone is running master, the transition will happen correctly. 

The diff looks bigger than it is, it's just moving the need bucket part of the code ahead.

### Testing

New unit test.

### CODE REVIEW

File: lib/db/leveldb_transactions.go [hunk 1]
```diff
@@ -149,6 +123,29 @@ func (t readWriteTransaction) updateGlobal(gk, folder, device []byte, file proto
...
+		oldGlobalFV = removedFV
+	} else if len(fl.Versions) > 1 {
+		// The previous newest version is now at index 1
+		oldGlobalFV = fl.Versions[1]
+	}
+	if oldFile, ok := t.getFile(folder, oldGlobalFV.Device, name); ok {
+		// A failure to get the file here is surprising and our
+		// global size data will be incorrect until a restart...
+		meta.removeFile(globalDeviceID, oldFile)
+	}
```
Comment (AudriusButkevicius) [inline lib/db/leveldb_transactions.go]: Else panic to weed out the bugs?

File: lib/db/leveldb_transactions.go [hunk 2]
```diff
@@ -149,6 +123,29 @@ func (t readWriteTransaction) updateGlobal(gk, folder, device []byte, file proto
 		t.Delete(nk)
 	}
 
+	if removedAt != 0 && insertedAt != 0 {
+		l.Debugf(`new global for "%v" after update: %v`, file.Name, fl)
+		t.Put(gk, mustMarshal(&fl))
```
Comment (AudriusButkevicius) [inline lib/db/leveldb_transactions.go]: Why is the delete gone?

File: lib/db/leveldb_transactions.go (hunk 1 above)
Comment (imsodin) [reply lib/db/leveldb_transactions.go]: I agree, but lets do that in a different commit to separate the potential ensuing madness from this fix :)

File: lib/db/leveldb_transactions.go (hunk 2 above)
Comment (imsodin) [reply lib/db/leveldb_transactions.go]: All need bucket operations, including the edge case that was handled at this point before, are now handled in the block that is above (https://github.com/imsodin/syncthing/blob/6e2b90e7701b1ec86a3c4afb758d0afe9d674532/lib/db/leveldb_transactions.go#L113)

**Pull request merged**